src/main/controlMod.F90

```diff
@@ -1113,11 +1116,50 @@ subroutine apply_use_init_interp(finidat_interp_dest, finidat, finidat_interp_so
        call endrun(msg=' ERROR: With use_init_interp, cannot use the same filename for source and dest')
     end if
 
-    finidat_interp_source = finidat
-    finidat = ' '
+    inquire(file=trim(finidat_interp_dest), exist=lexists)
+    if (lexists) then
+       ! open the input file and check for the name of the input source file
+       status = nf90_open(trim(finidat_interp_dest), 0, ncid)  
+       if (status /= nf90_noerr) call handle_err(status)
+       status = nf90_get_att(ncid, NF90_GLOBAL, 'initial_source_file', initial_source_file)
+       if (status /= nf90_noerr) call handle_err(status)
+       status = nf90_close(ncid)
+       if (status /= nf90_noerr) call handle_err(status)
+
+       ! If the input source file in finidat_interp_dest does not match the namelist input for mapping
+       ! then use the namelist input (i.e. finidat) - otherwise just use the generated file
+       if (trim(initial_source_file) /= trim(finidat)) then
+          finidat_interp_source = finidat
+          finidat = ' '
+          if (masterproc) then
```

Comment:
- [ ] [OPTIONAL] Here and a little below: this function is only called by the masterproc, so these masterproc conditionals aren't needed. They don't hurt, though they threw me off because it made me think that the routine was called by all procs – which I think would be somewhat problematic in that it would mean that all procs are opening the netcdf file to read the global attribute at once.

Reply:
@billsacks - thanks for pointing this out. I did not appreciate that this function was only called on masterproc. That said - I think there is a real problem in having an endrun call in a function that is only called on masterproc.
If you did encounter an issue - I think that the model would hang - since only the master processor would actually call endrun. So I think parts of controlMod need to be rewritten so that an error is broadcast to all processors and then the model can exit gracefully. I'm happy to implement that - but its a bit outside the scope of this PR. Thoughts?

Reply:
(We discussed this. It's fine to call endrun from one proc.)

Reply:
@billsacks - thanks for pointing this out. You are right!